### PR TITLE
CLDR-15404 add checks and tests

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-15404

common/main/root.xml
Change values in root to be 'empty' so that the tests force people to come up with their own

tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckCLDR.java
Add a new test, and new test failure type

tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPersonNames.java
Add a new check that will be run in the survey tool
* checks which sample names have to have certain field values
* checks which sample names have to not have certain field values
* checks that if surname2 is present, surname has to be
* checks that all multiword values have to have a space
* 
tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java
Add constant ALL for the SampleTypes
Add to interface new public getModifiedFieldToValue(); so we can access values without converting to different structure
Add some javadoc while we're at it

tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/SimpleNameObject.java
Add getModifiedFieldToValue()

tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java
Add TestLiteralCollapsing1() and TestLiteralCollapsing2() to test field suppression.
Remove TestNameSamples (since tests are moved to checks, so that vetters benefit from them)

- [ ] This PR completes the ticket.

### CODE REVIEW

**Review by pedberg-icu — APPROVED** [review]

**Review by pedberg-icu — COMMENTED** [review]
ConsoleCheck tests are failing for all locales except English because they are now inheriting empty sample name data, and there is a check that various sample name fields must not be empty.

**Comment by pedberg-icu** [comment]
> I made a quick fix to downgrade the missing items to warnings. We should refine after discussion, but this should now be safe to check in.

Well, there are still errors, see below for example. You can run ConsoleCheck before committing to see if the errors are fixed.
```
/home/runner/work/cldr/cldr/common/main/root.xml:5522:43: error
[31](https://github.com/unicode-org/cldr/runs/6097879845?check_suite_focus=true#step:6:31)
aa [Afar]	error	▸Miscellaneous|Person_Name_Formats|SampleName_Fields_for_Item:_givenSurname|given◂	〈Katherine〉	【】	〈∅∅∅〉	«=»	【】	⁅invalid place holder⁆	❮Error: Names must have a value for the ‘given‘ field. Mononyms (like ‘Lady Gaga’) use given, not surname❯	https://st.unicode.org/cldr-apps/v#/aa//20d63f048af4931a	root
[32](https://github.com/unicode-org/cldr/runs/6097879845?check_suite_focus=true#step:6:32)
Error:  Error: Names must have a value for the ‘given‘ field. Mononyms (like ‘Lady Gaga’) use given, not surname
[33](https://github.com/unicode-org/cldr/runs/6097879845?check_suite_focus=true#step:6:33)
/home/runner/work/cldr/cldr/common/main/root.xml:5531:43: error
[34](https://github.com/unicode-org/cldr/runs/6097879845?check_suite_focus=true#step:6:34)
aa [Afar]	error	▸Miscellaneous|Person_Name_Formats|SampleName_Fields_for_Item:_given2Surname|given◂	〈Alberto〉	【】	〈∅∅∅〉	«=»	【】	⁅invalid place holder⁆	❮Error: Names must have a value for the ‘given‘ field. Mononyms (like ‘Lady Gaga’) use given, not surname❯	https://st.unicode.org/cldr-apps/v#/aa//500a8a0bbf9f0b8f	root
[35](https://github.com/unicode-org/cldr/runs/6097879845?check_suite_focus=true#step:6:35)
Error:  Error: Names must have a value for the ‘given‘ field. Mononyms (like ‘Lady Gaga’) use given, not surname
[36](https://github.com/unicode-org/cldr/runs/6097879845?check_suite_focus=true#step:6:36)
```

**Review by pedberg-icu — APPROVED** [review]
Looks good now, thanks

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java [hunk 1]
```diff
@@ -101,6 +100,38 @@ public void TestBasic() {
...
+        "order=surnameFirst",   "❶{surname-allCaps}⓶ ❷{surname2}⓷ ❸{prefix}⓸ ❹{given}⓹ ❺{given2}⓺ ❻{suffix}❼",
+        "order=sorting",        "❶{surname}⓶ ❷{surname2}, ❸{prefix}⓸ ❹{given}⓹ ❺{given2}⓺ ❻{suffix}❼");
+
+    PersonNameFormatter personNameFormatter2 = new PersonNameFormatter(namePatternData2, FALLBACK_FORMATTER);
+
+    check(personNameFormatter2, sampleNameObject5, "order=givenFirst", "Mary⓷ ❹Smith"); // reasonable
+    check(personNameFormatter2, sampleNameObject5, "order=surnameFirst", "❶SMITH⓶ ❹Mary"); // reasonable
+    check(personNameFormatter2, sampleNameObject5, "order=sorting", "❶Smith⓶ ❹Mary"); // should be something like Smith,❸Mary
+}
+
```
Comment (richgillam) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java:134]: These two new tests (TestLiteralCollapsing1() and TestLiteralCollapsing2() ) duplicate my own test (TestLiteralTextElision() ) below.  I think you're doing some things I'm not, but I think it'd be good to agree on a name, get rid of any redundant test code, and put all of the tests closer together in the source file.

**Pull request merged**